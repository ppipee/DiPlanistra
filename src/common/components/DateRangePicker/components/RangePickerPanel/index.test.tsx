import React from 'react'

import 'jest-styled-components'
import { render } from '@testing-library/react'

import * as RangePickerPanelType from '.'

describe('Test RangePickerPanel', () => {
	const useDateRangeContextSpy = jest.fn()
	const overlaySpy = jest.fn()
	const closeSpy = jest.fn()

	jest.doMock('common/components/DateRangePicker/context', () => ({
		useDateRangeContext: useDateRangeContextSpy,
	}))

	jest.doMock('common/components/Gap', () => (props: Record<string, any>) => <div {...props}>{props.children}</div>)

	jest.doMock('common/components/Calendar', () => (props: Record<string, any>) => (
		<div {...props}>{props.children}</div>
	))

	jest.doMock('common/components/Overlay', () => (props: Record<string, any>) => {
		overlaySpy(props)

		return <div {...props}>{props.children}</div>
	})

	jest.doMock(
		'common/components/DateRangePicker/components/RangePickerPanelHeader',
		() => (props: Record<string, any>) => <div {...props}>{props.children}</div>,
	)

	const { default: RangePickerPanel } = require('.') as typeof RangePickerPanelType

	afterEach(() => {
		jest.resetAllMocks()
	})

	it('should render RangePickerPanel correctly', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: {
				startDate: new Date(2020, 1, 1),
				endDate: new Date(2020, 1, 1),
			},
		})

		const { container } = render(<RangePickerPanel>children</RangePickerPanel>)

		expect(useDateRangeContextSpy).toBeCalledTimes(1)
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should close DatePicker if overlay was clicked', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: {
				startDate: new Date(2020, 1, 1),
				endDate: new Date(2020, 1, 1),
			},
		})

		render(<RangePickerPanel />)

		overlaySpy.mock.calls[0][0].onClick()

		expect(closeSpy).toBeCalledTimes(1)
		expect(useDateRangeContextSpy).toBeCalledTimes(1)
	})
})
