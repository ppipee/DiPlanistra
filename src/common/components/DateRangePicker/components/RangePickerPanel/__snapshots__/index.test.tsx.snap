// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RangePickerPanel should render RangePickerPanel correctly 1`] = `
.c0 {
  background-color: #FFFFFF;
  padding: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 1px solid #BEC1C8;
  box-shadow: 0px 2px 8px -2px rgba(51,51,51,0.16);
  margin-top: 12px;
  position: absolute;
  z-index: 2;
}

<div
  class="c0"
>
  <div>
    RangePickerController
  </div>
  <div>
    <div />
    <div
      defaultdate="[object Object]"
    >
      children
    </div>
  </div>
</div>
`;
